azurerm/internal/services/mssqlvm/resource_arm_mssql_virtual_machine.go

```diff
@@ -0,0 +1,559 @@
+)
+
+func resourceArmMsSqlVirtualMachine() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMsSqlVirtualMachineCreateUpdate,
+		Read:   resourceArmMsSqlVirtualMachineRead,
+		Update: resourceArmMsSqlVirtualMachineCreateUpdate,
+		Delete: resourceArmMsSqlVirtualMachineDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"virtual_machine_resource_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					_, err := azure.ParseAzureResourceID(v)
+					errs = append(errs, err)
+					return
+				},
+			},
+
+			"sql_server_license_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sqlvirtualmachine.PAYG),
+					string(sqlvirtualmachine.AHUB),
+				}, false),
+			},
+
+			"sql_image_sku": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sqlvirtualmachine.Developer),
+					string(sqlvirtualmachine.Express),
+					string(sqlvirtualmachine.Standard),
+					string(sqlvirtualmachine.Enterprise),
+					string(sqlvirtualmachine.Web),
+				}, false),
+			},
+
+			"sql_virtual_machine_group_resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					_, err := azure.ParseAzureResourceID(v)
+					errs = append(errs, err)
+					return
+				},
+			},
+
+			"auto_patching_settings": {
+				Type:       schema.TypeList,
+				Optional:   true,
+				MaxItems:   1,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"day_of_week": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(sqlvirtualmachine.Monday),
+								string(sqlvirtualmachine.Tuesday),
+								string(sqlvirtualmachine.Wednesday),
+								string(sqlvirtualmachine.Thursday),
+								string(sqlvirtualmachine.Friday),
+								string(sqlvirtualmachine.Saturday),
+								string(sqlvirtualmachine.Sunday),
+							}, false),
+						},
+						"enable": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"maintenance_window_duration": {
```

Comment:
Could we append the units here, for example `maintenance_window_duration_in_seconds`